api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -238,6 +239,33 @@ message HttpConnectionManager {
     google.protobuf.BoolValue enabled = 3;
   }
 
+  // [#not-implemented-hide] Transformations that apply to path headers. Transformations are applied
+  // before any processing of requests by HTTP filters, routing, and matching. Only the normalized
+  // path will be visible internally if a transformation is enabled. Any path rewrites that the
+  // router performs (e.g. :ref:`regex_rewrite
+  // <envoy_api_field_config.route.v3.RouteAction.regex_rewrite>` or :ref:`prefix_rewrite
+  // <envoy_api_field_config.route.v3.RouteAction.prefix_rewrite>`) will apply to the *:path* header
+  // destined for the upstream.
+  //
+  // Note: access logging and tracing will show the original *:path* header.
```

Comment:
I wonder if we also need to add a way to log normalized path header. This is outside of the scope of this PR, but maybe add an Issue to track this.

Reply:
Yes, I imagine we can add a formatter `%UPSTREAM_PATH%` so users can see the path forwarded, and something to log the internal path used for matching. @chaoqin-li1123 FYI.

I added a comment here https://github.com/envoyproxy/envoy/issues/6589#issuecomment-786665159
when implementation starts we should file a followup